app/services/katello/candlepin/consumer.rb

```diff
@@ -120,6 +120,14 @@ def compliance_reasons
         self.class.friendly_compliance_reasons(Resources::Candlepin::Consumer.compliance(uuid)['reasons'])
       end
 
+      def entitlements?
+        # use cahced consumer_attributes if possible
+        count = @consumer_attributes.try(:[], 'entitlementCount')
+        return count > 0 if count
+
+        !entitlements.empty?
+      end
+
```

Comment:
This doesn't seem to be used anywhere in this PR.  Did it get refactored out on your last update?

Reply:
I am using it in [status calculation](https://github.com/Katello/katello/pull/7472/files#diff-eff3ce936eaeca07c83b52b05e8dee47R41).
The reason for this is to use cached version of `consumer_attributes` that I already have in [`update_subscription_facet`](https://github.com/Katello/katello/pull/7472/files#diff-bf897becee6d218f2e9b589c5f66dcfdR109). This way we are not sending N+1 requests because of status calculation.

Reply:
oh whoops!  not sure how i missed that :)